File: atm cajero sockets/cajero-cliente/src/atm/interfaces.py
from abc import ABC, abstractmethod
from typing import Optional, Tuple


class ICajeroServidor(ABC):

    @abstractmethod
    def depositar(self, numero_cuenta: str, clave: str, monto: float) -> Tuple[bool, str, Optional[float]]:
        raise NotImplementedError

    @abstractmethod
    def retirar(self, numero_cuenta: str, clave: str, monto: float) -> Tuple[bool, str, Optional[float]]:
        raise NotImplementedError

    @abstractmethod
    def consultar(self, numero_cuenta: str, clave: str) -> Tuple[bool, str, Optional[float]]:
        raise NotImplementedError

File: atm cajero sockets/cajero-cliente-go/src/atm/interfaces.go
// Equivalente a atm/interfaces.py: ICajeroServidor. En Go la interfaz es
// del lenguaje directamente, sin necesidad de ABC.
package main

type ICajeroServidor interface {
	Depositar(numeroCuenta, clave string, monto float64) (bool, string, *float64)
	Retirar(numeroCuenta, clave string, monto float64) (bool, string, *float64)
	Consultar(numeroCuenta, clave string) (bool, string, *float64)
}
